packages/core/lib/debug/printer.js

```diff
@@ -171,13 +171,31 @@ class DebugPrinter {
     const colorizedLines = splitLines(colorizedSource);
 
     this.config.logger.log("");
+    // We clone range though only "lines" is used here.
+    let rangePrintout = JSON.parse(JSON.stringify(range));
+    // We print a warning message and display the end of source code when the
+    // instruction's byte-offset to the start of the range in the source code
+    // is past the end of source code
+    const instruction = this.session.view(solidity.current.instruction);
+    if (instruction.start >= source.length) {
+      this.config.logger.log(
+        `${colors.bold("Warning:")} Past end of source, displaying end.`
```

Comment:
Hm, so I know I was the one who suggested this message to you, but I'm realizing now that when I did so, I forgot about the case where the location passed in isn't the current location.  Maybe change "Past end of source" to "Location is past end of source"?  To my mind, the former reads as necessarily being about the current location, whereas the latter doesn't, so I'd suggest changing that.